scripts/quests/outlands/You_Call_That_a_Knife.lua

```diff
@@ -0,0 +1,211 @@
+        end,
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Mhebi_Juhbily'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 0 and
+                        npcUtil.tradeHasExactly(trade, { xi.item.SANDFISH })
+                    then
+                        return quest:progressEvent(127, 0, xi.item.SANDFISH)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:event(318)
+                    end
+                end
+            },
+
+            ['Pula_Rhatti'] = quest:event(269, 0, xi.item.SANDFISH):replaceDefault(),
+
+            ['_6y9'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(128)
+                    end
+                end
+            },
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    if option == 511 then
+                        quest:setVar(player, 'Prog', 1)
+                        player:confirmTrade()
+                    end
+                end,
+
+                [128] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+                not player:hasKeyItem(xi.ki.NONBERRYS_KNIFE)
+        end,
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Mhebi_Juhbily'] = quest:event(129):replaceDefault(),
+
+            ['Pula_Rhatti'] = quest:event(269, 0, xi.item.SANDFISH):replaceDefault(),
+
+            ['Vah_Keshura'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(130)
+                    else
+                        return quest:event(131):replaceDefault()
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [130] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end
+            },
+        },
+
+        [xi.zone.TEMPLE_OF_UGGALEPIH] =
+        {
+            ['Chef_Nonberry'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { xi.item.TONBERRY_BOARD }) then
+                        return quest:progressEvent(27)
+                    else
+                        npcUtil.confirmTradedItems(trade) -- Take all items traded
```

Comment:
Why create a function for this? You should be able to just use player:confirmTrade()

Reply:
From my understanding `player:confirmTrade()` works on items that have been confirmed (such as through npcUtil.tradeHasExactly), I did not see a utility that would accept all items regardless of confirmation so I made this one.

Reply:
I'll leave that discretion with LSB mods.